src/pint/gridutils.py

```diff
@@ -1,16 +1,400 @@
 """Tools for building chi-squared grids."""
+import concurrent.futures
 import copy
+import logging
 import multiprocessing
 import os
-from multiprocessing import Process, Queue
 
 import astropy.constants as const
 import astropy.units as u
 import numpy as np
 
+try:
+    from tqdm import tqdm
+except ModuleNotFoundError:
+    tqdm = None
+
 import pint.utils
 
-__all__ = ["grid_chisq", "grid_chisq_mp", "plot_grid_chisq"]
+log = logging.getLogger(__name__)
+
+__all__ = ["doonefit", "grid_chisq", "grid_chisq_derived", "plot_grid_chisq"]
+
+
+def doonefit(ftr, parnames, parvalues):
+    """Worker process that computes one fit with specified parameters fixed
+
+    Parameters
+    ----------
+    ftr : pint.fitter.Fitter
+    parnames : list
+        Names of the parameters to grid over
+    parvalues : list
+        List of parameter values to grid over (each should be 1D array of astropy.units.Quantity)
+    """
+    # Make a full copy of the fitter to work with
+    myftr = copy.deepcopy(ftr)
+    for parname, parvalue in zip(parnames, parvalues):
+        # Freeze the  params we are going to grid over and set their values
+        # All other unfrozen parameters will be fitted for at each grid point
+        getattr(myftr.model, parname).frozen = True
+        getattr(myftr.model, parname).quantity = parvalue
+    myftr.fit_toas()
+    return myftr.resids.chi2
+
+
+def grid_chisq(
+    ftr, parnames, parvalues, executor=None, ncpu=None, chunksize=1, printprogress=True,
+):
+    """Compute chisq over a grid of parameters
+
+    Parameters
+    ----------
+    ftr : pint.fitter.Fitter
+        The base fitter to use.
+    parnames : list
+        Names of the parameters to grid over
+    parvalues : list
+        List of parameter values to grid over (each should be 1D array of :class:`astropy.units.Quantity`)
+    executor : concurrent.futures.Executor or None, optional
+        Executor object to run multiple processes in parallel
+        If None, will use default :class:`concurrent.futures.ProcessPoolExecutor`, unless overridden by ``ncpu=1``
+    ncpu : int, optional
+        If an existing Executor is not supplied, one will be created with this number of workers.
+        If 1, will run single-processor version
+        If None, will use :func:`multiprocessing.cpu_count`
+    chunksize : int
```

Comment:
What does this actually do? Since users are passing in existing executors does it do anything?